src/cascadia/TerminalApp/MinMaxCloseControl.cpp

```diff
@@ -11,29 +11,41 @@
 
 namespace winrt::TerminalApp::implementation
 {
-    MinMaxCloseControl::MinMaxCloseControl(uint64_t hWnd) :
-        _window(reinterpret_cast<HWND>(hWnd))
+    MinMaxCloseControl::MinMaxCloseControl()
     {
         const winrt::Windows::Foundation::Uri resourceLocator{ L"ms-appx:///MinMaxCloseControl.xaml" };
         winrt::Windows::UI::Xaml::Application::LoadComponent(*this, resourceLocator, winrt::Windows::UI::Xaml::Controls::Primitives::ComponentResourceLocation::Nested);
     }
 
+    uint64_t MinMaxCloseControl::ParentWindowHandle() const
+    {
+        return reinterpret_cast<uint64_t>(_window.get());
+    }
+
+    void MinMaxCloseControl::ParentWindowHandle(uint64_t handle)
+    {
+        _window.reset(reinterpret_cast<HWND>(handle));
+    }
+
     void MinMaxCloseControl::_OnMaximize(byte flag)
     {
-        POINT point1 = {};
-        ::GetCursorPos(&point1);
-        const LPARAM lParam = MAKELPARAM(point1.x, point1.y);
-        WINDOWPLACEMENT placement = { sizeof(placement) };
-        ::GetWindowPlacement(_window, &placement);
-        if (placement.showCmd == SW_SHOWNORMAL)
-        {
-            winrt::Windows::UI::Xaml::VisualStateManager::GoToState(this->Maximize(), L"WindowStateMaximized", false);
-            ::PostMessage(_window, WM_SYSCOMMAND, SC_MAXIMIZE | flag, lParam);
-        }
-        else if (placement.showCmd == SW_SHOWMAXIMIZED)
+        if (_window)

```

Comment:
unrelated: github's diff is just ![](https://i.imgur.com/ML6ZlAI.gif)

Reply:
Cool tip: append `?w=1` to the end of the URL:

![image](https://user-images.githubusercontent.com/14316954/60907749-2639de00-a22f-11e9-9a13-39dfcae9e8bb.png)

It elides most horizontal whitespace changes!

Reply:
_why can't I set that to **always**_